apps/dashboard/src/components/subscribers/timezone-select.tsx

```diff
@@ -1,69 +1,102 @@
 
   return (
-    <Select
-      value={value}
-      onValueChange={(val) => {
-        const parsedValue = parseTimezone(val);
-        onValueChange(parsedValue.value);
-      }}
-      disabled={disabled || readOnly}
-      required={required}
-      defaultValue={defaultOption}
-    >
-      <SelectTrigger className="focus:ring-stroke-strong group overflow-hidden p-1.5 shadow-sm focus:ring-1">
-        <SelectValue
-          placeholder={
-            <div className="flex w-full items-center gap-1">
-              <div>
-                <RiTimeLine className="size-4" />
-              </div>
-              <TruncatedText className="text-sm">Select a timezone</TruncatedText>
-            </div>
-          }
-          asChild
-          className="w-full overflow-hidden"
+    <Popover open={open} onOpenChange={setOpen}>
+      <PopoverTrigger asChild>
+        <Button
+          type="button"
+          variant="secondary"
+          mode="outline"
+          className="flex h-8 w-full items-center gap-1 rounded-lg px-3 focus:z-10"
+          disabled={disabled}
         >
           <div className="flex max-w-full flex-1 items-center gap-1 overflow-hidden">
             <div>
               <RiTimeLine className="size-4" />
             </div>
-            {value && (
-              <TruncatedText className="text-foreground w-full min-w-0 flex-1 text-sm">
+            {value ? (
+              <TruncatedText className="text-foreground w-full min-w-0 flex-1 text-xs font-normal text-neutral-950">
                 {parseTimezone(value).label}
               </TruncatedText>
+            ) : (
+              <TruncatedText className="w-full min-w-0 flex-1 text-xs font-normal text-neutral-400">
+                Search timezone...
+              </TruncatedText>
             )}
+            <RiArrowDownSLine
+              className={cn('ml-auto size-4 opacity-50', disabled || readOnly ? 'hidden' : 'opacity-100')}
+            />
           </div>
-        </SelectValue>
-      </SelectTrigger>
-      <SelectContent>
-        {options.map((item) => (
-          <SelectItem key={item.value} value={item.value}>
-            {item.label}
-          </SelectItem>
-        ))}
-      </SelectContent>
-    </Select>
+        </Button>
+      </PopoverTrigger>
+      <PopoverContent className="w-[300px] rounded-lg border-t-0 p-0">
+        <Command>
+          <CommandInput
+            placeholder="Search timezone..."
+            /**
+             * Scroll to top bug workaround: https://github.com/pacocoursey/cmdk/issues/233#issuecomment-2015998940
+             */
+            onValueChange={() => {
+              // clear pending scroll
+              clearTimeout(scrollId.current);
+
+              // the setTimeout is used to create a new task
+              // this is to make sure that we don't scroll until the user is done typing
+              // you can tweak the timeout duration ofc
+              scrollId.current = setTimeout(() => {
```

Comment:
I am curious if [scrollIntoView](https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollIntoView) helps with this.

Reply:
Had tried that It didnt work